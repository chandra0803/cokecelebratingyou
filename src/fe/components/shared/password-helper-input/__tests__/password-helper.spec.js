import React from 'react';
import PasswordHelper from '../password-helper.js';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

const mockOptions = {
    name: 'newPassword',
    label: 'New Password',
    value: 'Testing123!',
    disabled: false,
    tabIndex: '1',
    onChange: () => true,
    passwordRules: {
        maxLength: 16,
        minLength: 8,
        isLowerCaseRequired: true,
        isUpperCaseRequired: true,
        isNumberRequired: true,
        isSpecialCharRequired: true,
        hasValidLength: true
    },
    passwordLabels: {
        length: '',
        isLowerCaseRequired: '',
        isUpperCaseRequired: '',
        isNumberRequired: '',
        isSpecialCharRequired: ''
    }
};
const mockOptions2 = {
    name: 'newPassword',
    label: 'New Password',
    value: 'Testing123!',
    disabled: false,
    tabIndex: '1',
    onChange: () => true,
    passwordRules: {
        isLowerCaseRequired: false,
        isUpperCaseRequired: false,
        isSpecialCharRequired: false,
        hasValidLength: false
    },
    passwordLabels: {
        length: '',
        isLowerCaseRequired: '',
        isUpperCaseRequired: '',
        isNumberRequired: '',
        isSpecialCharRequired: ''
    }
};
test( 'Password Helper renders different rules depending on props passed in', () => {
    const component = renderer.create(
        <PasswordHelper label="New Password"
                        name= "newPassword"
                        value= { mockOptions.value }
                        tabIndex="2"
                        onChange={ () => true }
                        passwordRules={ mockOptions.passwordRules }
                        passwordLabels={ mockOptions.passwordLabels } />
    );

    const tree = component.toJSON();
    expect( tree ).toMatchSnapshot();
} );
test( 'Password Helper renders different rules depending on props passed in', () => {
    const component = renderer.create(
        <PasswordHelper label="New Password"
                        name= "newPassword"
                        value= { mockOptions2.value }
                        tabIndex="2"
                        onChange={ () => true }
                        passwordRules={ mockOptions2.passwordRules }
                        passwordLabels={ mockOptions2.passwordLabels } />
    );

    const tree = component.toJSON();
    expect( tree ).toMatchSnapshot();
} );
it ( 'changes focused value in state when password input is focused  ', () => {
    const props = {
        ...mockOptions
    };

    const tree = mount( <PasswordHelper { ...props } /> );

    const input = tree.find( 'input[name="newPassword"]' );

    expect ( tree.state( 'focused' ) ).toBe( false );

    input.simulate( 'focus', { target: { name: input.props().name } } );

    tree.setState( { focused: true } );

    expect ( tree.state( 'focused' ) ).toBe( true );
} );

it ( 'changes focused value in state when password input is blurred ', () => {
    const props = {
        ...mockOptions
    };

    const tree = mount( <PasswordHelper { ...props } /> );

    const input = tree.find( 'input[name="newPassword"]' );

    expect ( tree.state( 'focused' ) ).toBe( false );

    input.simulate( 'blur', { target: { name: input.props().name } } );

    tree.setState( { focused: true } );

    expect ( tree.state( 'focused' ) ).toBe( true );
} );
it( 'executes lengthTest and returns true or false, depending on data passed in', ()=>{
    const props = {
        ...mockOptions
    };
    const tree = mount( <PasswordHelper { ...props } /> );

    const invalid = tree.instance().lengthTest( 'password', 10 );

    expect( invalid ).toBe( false );

    const valid = tree.instance().lengthTest( 'password', 6 );

    expect( valid ).toBe( true );
} );
