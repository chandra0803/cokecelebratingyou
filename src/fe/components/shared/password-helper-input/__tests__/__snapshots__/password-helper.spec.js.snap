// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password Helper renders different rules depending on props passed in 1`] = `
<div
  className="password-helper-input"
>
  <div
    className="password-input"
  >
    <div
      className="input-wrap"
    >
      <label
        className="control-label"
        htmlFor="password1"
      >
        New Password
      </label>
      <span
        className="eye"
        onClick={[Function]}
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,3.9C4.2,3.9,0,9.6,0,12s4.2,8.1,12,8.1s12-5.8,12-8.1S19.8,3.9,12,3.9z M12,18.7c-7.2,0-10.6-5.4-10.6-6.7 S4.8,5.3,12,5.3s10.6,5.4,10.6,6.7S19.2,18.7,12,18.7z"
          />
          <path
            d="M12,7c-2.8,0-5,2.2-5,5s2.2,5,5,5s5-2.2,5-5S14.8,7,12,7z M12,15.6c-2,0-3.6-1.6-3.6-3.6c0-2,1.6-3.6,3.6-3.6 c2,0,3.6,1.6,3.6,3.6C15.6,14,14,15.6,12,15.6z"
          />
        </svg>
      </span>
      <input
        autoFocus={undefined}
        className="gq-input"
        disabled={undefined}
        name="newPassword"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyPress={undefined}
        tabIndex="2"
        type="text"
        value="Testing123!"
      />
    </div>
  </div>
  <div
    className="password-rules-container hide"
  >
    <div
      className="rule"
    >
      <div
        className="hide"
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
          />
          <path
            d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
          />
        </svg>
      </div>
      <div
        className="hide"
      >
        <span>
          <svg
            className="icon"
            viewBox="0 0 24 24"
          >
            <path
              d="M14.6,8.3l-2.7,2.7L9.3,8.3C9,8,8.5,8,8.2,8.3c-0.3,0.3-0.4,0.8,0,1.1l2.7,2.7l-2.5,2.5C8,15,8,15.5,8.3,15.8 c0.3,0.3,0.8,0.4,1.1,0l2.7-2.7l2.7,2.7c0.3,0.3,0.8,0.4,1.1,0c0.3-0.3,0.4-0.8,0-1.1L13.3,12L16,9.3c0.3-0.3,0.4-0.8,0-1.1 C15.4,8,14.9,8,14.6,8.3z M3.5,3.5c-4.7,4.7-4.7,12.3,0,17s12.3,4.7,17,0s4.7-12.3,0-17S8.2-1.2,3.5,3.5z M19.4,19.4 c-4.1,4.1-10.7,4.1-14.8,0S0.5,8.7,4.6,4.6s10.7-4.1,14.8,0S23.5,15.3,19.4,19.4z"
            />
          </svg>
        </span>
      </div>
      <span>
        
      </span>
    </div>
    <div
      className="rule"
    >
      <div
        className="hide"
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
          />
          <path
            d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
          />
        </svg>
      </div>
      <div
        className="rule-icon close-circle-icon"
      >
        <span>
          <svg
            className="icon"
            viewBox="0 0 24 24"
          >
            <path
              d="M14.6,8.3l-2.7,2.7L9.3,8.3C9,8,8.5,8,8.2,8.3c-0.3,0.3-0.4,0.8,0,1.1l2.7,2.7l-2.5,2.5C8,15,8,15.5,8.3,15.8 c0.3,0.3,0.8,0.4,1.1,0l2.7-2.7l2.7,2.7c0.3,0.3,0.8,0.4,1.1,0c0.3-0.3,0.4-0.8,0-1.1L13.3,12L16,9.3c0.3-0.3,0.4-0.8,0-1.1 C15.4,8,14.9,8,14.6,8.3z M3.5,3.5c-4.7,4.7-4.7,12.3,0,17s12.3,4.7,17,0s4.7-12.3,0-17S8.2-1.2,3.5,3.5z M19.4,19.4 c-4.1,4.1-10.7,4.1-14.8,0S0.5,8.7,4.6,4.6s10.7-4.1,14.8,0S23.5,15.3,19.4,19.4z"
            />
          </svg>
        </span>
      </div>
      <span />
    </div>
    <div
      className="rule child-label"
    >
      <div
        className="hide"
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
          />
          <path
            d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
          />
        </svg>
      </div>
      <div
        className="rule-icon close-circle-icon"
      >
        <span
          className="circle"
        />
      </div>
      <span />
    </div>
    <div
      className="rule child-label"
    >
      <div
        className="hide"
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
          />
          <path
            d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
          />
        </svg>
      </div>
      <div
        className="hide"
      >
        <span
          className="circle"
        />
      </div>
      <span>
        
      </span>
    </div>
    <div
      className="rule child-label"
    >
      <div
        className="hide"
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
          />
          <path
            d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
          />
        </svg>
      </div>
      <div
        className="hide"
      >
        <span
          className="circle"
        />
      </div>
      <span>
        
      </span>
    </div>
    <div
      className="rule child-label"
    >
      <div
        className="hide"
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
          />
          <path
            d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
          />
        </svg>
      </div>
      <div
        className="hide"
      >
        <span
          className="circle"
        />
      </div>
      <span>
        
      </span>
    </div>
    <div
      className="rule child-label"
    >
      <div
        className="hide"
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
          />
          <path
            d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
          />
        </svg>
      </div>
      <div
        className="hide"
      >
        <span
          className="circle"
        />
      </div>
      <span>
        
      </span>
    </div>
    <div
      className="rule child-label"
    >
      <div
        className="hide"
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
          />
          <path
            d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
          />
        </svg>
      </div>
      <div
        className="hide"
      >
        <span
          className="circle"
        />
      </div>
      <span />
    </div>
  </div>
</div>
`;

exports[`Password Helper renders different rules depending on props passed in 2`] = `
<div
  className="password-helper-input"
>
  <div
    className="password-input"
  >
    <div
      className="input-wrap"
    >
      <label
        className="control-label"
        htmlFor="password1"
      >
        New Password
      </label>
      <span
        className="eye"
        onClick={[Function]}
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,3.9C4.2,3.9,0,9.6,0,12s4.2,8.1,12,8.1s12-5.8,12-8.1S19.8,3.9,12,3.9z M12,18.7c-7.2,0-10.6-5.4-10.6-6.7 S4.8,5.3,12,5.3s10.6,5.4,10.6,6.7S19.2,18.7,12,18.7z"
          />
          <path
            d="M12,7c-2.8,0-5,2.2-5,5s2.2,5,5,5s5-2.2,5-5S14.8,7,12,7z M12,15.6c-2,0-3.6-1.6-3.6-3.6c0-2,1.6-3.6,3.6-3.6 c2,0,3.6,1.6,3.6,3.6C15.6,14,14,15.6,12,15.6z"
          />
        </svg>
      </span>
      <input
        autoFocus={undefined}
        className="gq-input"
        disabled={undefined}
        name="newPassword"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyPress={undefined}
        tabIndex="2"
        type="text"
        value="Testing123!"
      />
    </div>
  </div>
  <div
    className="password-rules-container hide"
  >
    <div
      className="rule"
    >
      <div
        className="hide"
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
          />
          <path
            d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
          />
        </svg>
      </div>
      <div
        className="hide"
      >
        <span>
          <svg
            className="icon"
            viewBox="0 0 24 24"
          >
            <path
              d="M14.6,8.3l-2.7,2.7L9.3,8.3C9,8,8.5,8,8.2,8.3c-0.3,0.3-0.4,0.8,0,1.1l2.7,2.7l-2.5,2.5C8,15,8,15.5,8.3,15.8 c0.3,0.3,0.8,0.4,1.1,0l2.7-2.7l2.7,2.7c0.3,0.3,0.8,0.4,1.1,0c0.3-0.3,0.4-0.8,0-1.1L13.3,12L16,9.3c0.3-0.3,0.4-0.8,0-1.1 C15.4,8,14.9,8,14.6,8.3z M3.5,3.5c-4.7,4.7-4.7,12.3,0,17s12.3,4.7,17,0s4.7-12.3,0-17S8.2-1.2,3.5,3.5z M19.4,19.4 c-4.1,4.1-10.7,4.1-14.8,0S0.5,8.7,4.6,4.6s10.7-4.1,14.8,0S23.5,15.3,19.4,19.4z"
            />
          </svg>
        </span>
      </div>
      <span>
        
      </span>
    </div>
    <div
      className="rule"
    >
      <div
        className="hide"
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
          />
          <path
            d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
          />
        </svg>
      </div>
      <div
        className="rule-icon close-circle-icon"
      >
        <span>
          <svg
            className="icon"
            viewBox="0 0 24 24"
          >
            <path
              d="M14.6,8.3l-2.7,2.7L9.3,8.3C9,8,8.5,8,8.2,8.3c-0.3,0.3-0.4,0.8,0,1.1l2.7,2.7l-2.5,2.5C8,15,8,15.5,8.3,15.8 c0.3,0.3,0.8,0.4,1.1,0l2.7-2.7l2.7,2.7c0.3,0.3,0.8,0.4,1.1,0c0.3-0.3,0.4-0.8,0-1.1L13.3,12L16,9.3c0.3-0.3,0.4-0.8,0-1.1 C15.4,8,14.9,8,14.6,8.3z M3.5,3.5c-4.7,4.7-4.7,12.3,0,17s12.3,4.7,17,0s4.7-12.3,0-17S8.2-1.2,3.5,3.5z M19.4,19.4 c-4.1,4.1-10.7,4.1-14.8,0S0.5,8.7,4.6,4.6s10.7-4.1,14.8,0S23.5,15.3,19.4,19.4z"
            />
          </svg>
        </span>
      </div>
      <span />
    </div>
    <div
      className="hide"
    >
      <div
        className="hide"
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
          />
          <path
            d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
          />
        </svg>
      </div>
      <div
        className="hide"
      >
        <span
          className="circle"
        />
      </div>
      <span>
        
      </span>
    </div>
    <div
      className="hide"
    >
      <div
        className="hide"
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
          />
          <path
            d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
          />
        </svg>
      </div>
      <div
        className="hide"
      >
        <span
          className="circle"
        />
      </div>
      <span>
        
      </span>
    </div>
    <div
      className="hide"
    >
      <div
        className="hide"
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
          />
          <path
            d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
          />
        </svg>
      </div>
      <div
        className="hide"
      >
        <span
          className="circle"
        />
      </div>
      <span>
        
      </span>
    </div>
    <div
      className="hide"
    >
      <div
        className="hide"
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
          />
          <path
            d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
          />
        </svg>
      </div>
      <div
        className="hide"
      >
        <span
          className="circle"
        />
      </div>
      <span />
    </div>
  </div>
</div>
`;
